command/agent/node_endpoint.go

```diff
@@ -205,7 +205,7 @@ func (s *HTTPServer) nodePurge(resp http.ResponseWriter, req *http.Request, node
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
 	args := structs.NodeDeregisterRequest{
```

Comment:
Since this change is intended to alter the api for the future and deprecate the old call, I'd rather add the new batch field to the request and mark the old as deprecated

Reply:
I've come back around to your opinion here, and the batch has been implemented as a new message